addons/l10n_es_edi_facturae/models/account_move.py

```diff
@@ -0,0 +1,352 @@
+        comodel_name='ir.attachment',
+        string="Facturae Attachment",
+        compute=lambda self: self._compute_linked_attachment_id('l10n_es_edi_facturae_xml_id', 'l10n_es_edi_facturae_xml_file'),
+        depends=['l10n_es_edi_facturae_xml_file']
+    )
+    l10n_es_edi_facturae_xml_file = fields.Binary(
+        attachment=True,
+        string="Facturae File",
+        copy=False,
+    )
+
+    def _convert_computed_tax_to_template(self, computed_tax_dict):
+        """ Helper to convert the tax dict from a _compute_taxes() into a dict usable in the template """
+        tax = self.env["account.tax"].browse(computed_tax_dict["tax_id"])
+        return {
+            "tax_record": tax,
+            "TaxRate": f'{abs(tax.amount):.3f}',
+            "TaxableBase": {
+                'TotalAmount': computed_tax_dict["base_amount_currency"],
+                'EquivalentInEuros': computed_tax_dict["base_amount"],
+            },
+            "TaxAmount": {
+                "TotalAmount": abs(computed_tax_dict["tax_amount_currency"]),
+                "EquivalentInEuros": abs(computed_tax_dict["tax_amount"]),
+            },
+        }
+
+    @api.model
+    def _l10n_es_edi_facturae_inv_lines_to_items(self):
+        """
+        Convert the invoice lines to a list of items required for the Facturae xml generation
+        :return:                        A tuple containing the Face items, the taxes and the invoice totals data.
+        """
+        self.ensure_one()
+        items = []
+        totals = {
+            'total_gross_amount': 0.,
+            'total_general_discounts': 0.,
+            'total_general_surcharges': 0.,
+            'total_taxes_withheld': 0.,
+            'total_tax_outputs': 0.,
+            'total_payments_on_account': 0.,
+            'amounts_withheld': 0.,
+        }
+        taxes = []
+        taxes_withheld = []
+        for line in self.invoice_line_ids:
+            invoice_line_values = {}
+
+            discount = max(0., (line.discount * line.price_unit / 100))
+            surcharge = abs(min(0., (line.discount * line.price_unit / 100)))
+            total_price = line.price_subtotal + discount - surcharge
+            gross_amount = line.price_subtotal
+            totals['total_gross_amount'] += gross_amount
+            totals['total_general_discounts'] += discount
+            totals['total_general_surcharges'] += surcharge
+            base_line = self.env['account.tax']._convert_to_tax_base_line_dict(
+                line, partner=line.partner_id, currency=line.currency_id, product=line.product_id, taxes=line.tax_ids,
+                price_unit=line.price_unit, quantity=line.quantity, discount=line.discount, account=line.account_id,
+                price_subtotal=line.price_subtotal, is_refund=line.is_refund, rate=line.amount_currency/line.balance if line.balance else 0.
+            )
+
+            taxes_computed = self.env['account.tax']._compute_taxes([base_line])
```

Comment:
you need to compute the taxes all at once...
```suggestion
            taxes_computed = self.env['account.tax']._compute_taxes(base_lines)
```